### ISSUE TITLE
Optimize kubermatic api

### ISSUE THREAD

**Issue by PratikDeoghare** [kubermatic/kubermatic#9302]
Some observations re KKP api. 

1. Each request creates new server object.
`   return httptransport.NewServer(`
  is in every handler. Don't think that is necessary.
	
2. Gorilla is the worst performing mux.
  
   It won't get better because 
	   - maps + sync.Pool don't play well together.
 	   - gorilla makes heavy use of maps

   Consider using alternatives  https://github.com/julienschmidt/go-http-routing-benchmark

3. Many possibly unnecessary requests for social media icons
  from dashboard.

4. There were panics in the logs. They need to investigated. 

If this is optimized, we can possibly serve many more
requests with current resources (possibly with fewer resources).

**Comment by xrstf** [comment]
Please copy-paste the Slack discussion in here, or else it's hard to follow what this ticket is all about.

**Comment by xrstf** [comment]
> Consider using alternatives https://github.com/julienschmidt/go-http-routing-benchmark

Please let's not bikeshed about microsecond differences in Go routers. Saving 0.0001% is not worth the often much worse Go API, features, stability etc.

"This router is a tiny teeensy bit faster than gorilla/mux." is IMHO not an argument to change it.
